### PR TITLE
Make Dockerfiles more copyable

### PR DESCRIPTION
**Pull request by sclarkson**
Remove the external file dependencies and reduce the code re-use by combining all Dockerfiles into a single file with a version ARG.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	DEBIAN_FRONTEND=noninteractive \
+		apt-get install \
+		--yes \
+		--no-install-recommends \
+		--option "Acquire::http::No-Cache=true" \
+		--option "Acquire::http::Pipeline-Depth=0" \
+		lambda-stack-cuda && \
+	rm -rf /var/lib/apt/lists/*
+
+# Let pip break system packages
```
Comment (Jordan-Lambda) [inline Dockerfile:88]: I don't think I want this in the dockerfile.

It feels out of place when the user is (hopefully) trying to be more methodical and intentional about their "system".

I'm still fine with the PR as a whole with this still in it, just wanted to comment while I notice it :)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+RUN printf "\
+[global]\n\
+break-system-packages = true\n\
+" > /etc/pip.conf
+
+# Setup for nvidia-docker
+ENV NVIDIA_VISIBLE_DEVICES all
+ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
+ENV NVIDIA_REQUIRE_CUDA "cuda>=12.4"
```
Comment (Jordan-Lambda) [inline Dockerfile:97]: Will this (intentionally) cause workloads in this container to fail if the host doesn't have a CUDA 12.4 or higher nvidia driver?

Again, not blocking me accepting this PR in any way.

**Review by Jordan-Lambda — APPROVED** [review]
I like this as a single self-contained dockerfile. Nice :slightly_smiling_face:

File: Dockerfile (hunk 2 above)
Comment (sclarkson) [reply Dockerfile:97]: It allegedly prevents the container from starting at all if the host CUDA version isn't sufficient, but I have never tried it.

https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/docker-specialized.html#nvidia-require-cuda-constraint

File: Dockerfile (hunk 1 above)
Comment (sclarkson) [reply Dockerfile:88]: I put it there to level the behavior between 22.04 and 24.04. There's also not any "danger" in unifying your pip and system python environments in a container, IMO.

**Pull request merged**